hack/dev-setup

```diff
@@ -0,0 +1,98 @@
+#!/bin/sh
+
+DEV_DIR=$(dirname "${0}")/../dev
+EXAMPLE_DIR=$(dirname "${0}")/../example
+MINIKUBE_IP=$(minikube ip)
+IP_ROUTE=$(ip route get 1)
+IP_ADDRESS=$(echo ${IP_ROUTE#*src} | awk '{print $1}')
+
+echo "After OSX sleep, minikube's time is off."
+echo "This results in x509 certificate auth to not work for shoot certificates"
+minikube ssh -- docker run -i --rm --privileged --pid=host debian nsenter -t 1 -m -u -n -i date -u $(date -u +%m%d%H%M%Y)
+
+mkdir -p ${DEV_DIR}
+
+cp ${EXAMPLE_DIR}/componentconfig-gardener-controller-manager.yaml ${DEV_DIR}/
+
+kubectl apply -f ${EXAMPLE_DIR}/namespace-garden.yaml
+kubectl apply -f ${EXAMPLE_DIR}/namespace-garden-core.yaml
+kubectl apply -f ${EXAMPLE_DIR}/secret-default-domain.yaml
+kubectl apply -f ${EXAMPLE_DIR}/secret-internal-domain.yaml
+
+cat <<EOF | kubectl apply -f -
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: etcd
+  namespace: garden
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      run: etcd
+  template:
+    metadata:
+      labels:
+        run: etcd
+    spec:
+      containers:
+      - image: quay.io/coreos/etcd:v3.1.10
+        name: etcd
+        command:
+        - etcd
+        - -advertise-client-urls=http://${MINIKUBE_IP}:2379
+        - -listen-client-urls=http://0.0.0.0:2379
```

Comment:
Is it easily possible to configure the `etcd` deployment so that it uses some host path for its data? If it would be too complex, let's not do it.

Reply:
I've added it